webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,256 @@
+        />
+    );
+
+    const notification = (
+        <NotificationBox
+            variant={'info'}
+            text={notificationText}
+            id={'move_thread'}
+        />
+    );
+
+    const handlePostError = (error: ClientError) => {
+        setIsButtonClicked(false);
+        setPostError(error.message);
+        setHasError(true);
+        setTimeout(() => setHasError(false), Constants.ANIMATION_TIMEOUT);
+    };
+
+    const handleSubmit = async () => {
+        setIsButtonClicked(true);
+        if (!selectedChannel) {
+            setIsButtonClicked(false);
+            return;
+        }
+
+        const channel = selectedChannel.details;
+
+        let result = await actions.moveThread(post.root_id || post.id, channel.id);
+
+        if (result.error) {
+            handlePostError(result.error);
+            return;
+        }
+
+        if (
+            channel.type === Constants.MENTION_MORE_CHANNELS &&
+            channel.type === Constants.OPEN_CHANNEL
+        ) {
+            result = await actions.joinChannelById(channel.id);
+
+            if (result.error) {
+                handlePostError(result.error);
+                return;
+            }
+        }
+
+        result = await actions.switchToChannel(channel);
+
+        if (result.error) {
+            handlePostError(result.error);
+            return;
+        }
+
+        onHide();
+    };
+
+    const postPreviewFooterMessage = formatMessage({
+        id: 'move_thread_modal.preview.footer_message',
+        defaultMessage: 'Originally posted in ~{channelName}',
+
+    },
+    {
+        channelName: originalChannel.display_name,
+    });
+
+    return (
+        <GenericModal
+            className='a11y__modal forward-post move-thread'
+            id='move-thread-modal'
+            show={true}
+            autoCloseOnConfirmButton={false}
+            compassDesign={true}
+            modalHeaderText={formatMessage({
+                id: 'move_thread_modal.title',
+                defaultMessage: 'Move thread',
+            })}
+            confirmButtonText={formatMessage({
+                id: 'move_thread_modal.button.forward',
+                defaultMessage: 'Move',
+            })}
+            cancelButtonText={formatMessage({
+                id: 'move_thread_modal.button.cancel',
+                defaultMessage: 'Cancel',
+            })}
+            isConfirmDisabled={isButtonClicked}
+            handleConfirm={handleSubmit}
+            handleEnterKeyPress={handleSubmit}
+            handleCancel={onHide}
+            onExited={onHide}
+        >
+            <div
+                className={'move-thread__body'}
+                ref={measuredRef}
+            >
+                {notification}
+                <ForwardPostChannelSelect
+                    onSelect={handleChannelSelect}
+                    value={selectedChannel}
+                    currentBodyHeight={bodyHeight}
+                    validChannelTypes={['O', 'P']}
```

Comment:
Are we not able to move messages from a public channel to a DM? I may have missed it, but I did not see anything explicitly prohibiting it at least on the server side.

Reply:
There are some intricacies (from an unread notification standpoint) with moving a thread from point A to B where the participants in point A aren't all members of point B. For now, we're going to filter out DMs and GMs, hoping to address this in a future iteration.